targets/TARGET_Freescale/TARGET_MCUXpresso_MCUS/TARGET_MCU_K64F/mbed_crc_api.c

```diff
@@ -0,0 +1,72 @@
+#include "crc_api.h"
+
+#include "drivers/fsl_crc.h"
+#include "platform/mbed_assert.h"
+
+#ifdef DEVICE_CRC
+
+static crc_bits_t width;
+static uint32_t final_xor;
+
+bool hal_crc_is_supported(const crc_mbed_config_t* config)
+{
+  if (config == NULL)
+    return false;
+
+  if ((config->width != 32) || (config->width != 16))
+    return false;
+
+  return true;
+}
+
+void hal_crc_compute_partial_start(const crc_mbed_config_t* config)
+{
+  if (config == NULL)
+    return;
+
+  width = ((config->polynomial & 0xFFFF0000U) != 0) ? kCrcBits32 : kCrcBits16;
```

Comment:
Why do we need this extra intelligence here? User passes the width, and it is available as part of config->width. We can simply use that.